egs/wsj/s5/steps/nnet3/multilingual/get_egs.sh

```diff
@@ -0,0 +1,158 @@
+#!/bin/bash
+
+# This script generates separate egs directory for each input 
+# language in multilingual setup, which contains both egs.*.ark and egs.*.scp.
+#
+# Then it uses separate egs directory for each language, 
+# to generate egs.*.scp file where the scp points to
+# original archive for each egs directory.
+# egs.*.scp is randomized w.r.t language id.
+#
+# Also it generates egs.JOB.scp, output.JOB.scp and weight.JOB.scp,
+# where output file contains language-id for each example
+# and weight file contains weights for scaling output posterior 
+# for each example w.r.t input language.
+#
+# This script will generally be called from nnet training script
+# and combine training examples used to train multilingual neural net.
+
+echo "$0 $@"  # Print the command line for logging
+. ./cmd.sh
+set -e
+
+
+# Begin configuration section
+cmd=run.pl
+stage=0
+get_egs_stage=0
+left_context=13
+right_context=9
+online_ivector_dir=     # can be used if we are including speaker information as iVectors.
+samples_per_iter=400000 # this is the target number of egs in each archive of egs
+                        # (prior to merging egs).  We probably should have called
+                        # it egs_per_iter. This is just a guideline; it will pick
+                        # a number that divides the number of samples in the
+                        # entire data.
+# Configuration to allocate egs
+minibatch_size=512
+num_archives=100
+num_jobs=10
+echo "$0 $@"  # Print the command line for logging
+
+if [ -f path.sh ]; then . ./path.sh; fi
+. parse_options.sh || exit 1;
+
+if [ $# -lt 4 ]; then
+  echo "Usage: $0 [opts] num-input-langs <data-dir-per-lang> <ali-dir-per-lang> <egs-dir-per-lang> <multilingual-egs-dir>"
```

Comment:
I think it would be better to just take as input the egs dirs from all the languages, assuming they already exist; and move the responsibility for dumping those egs to a further-out level.